common/src/api/external/error.rs

```diff
@@ -156,6 +160,18 @@ impl Error {
     pub fn unavail(message: &str) -> Error {
         Error::ServiceUnavailable { internal_message: message.to_owned() }
     }
+
+    /// Generates an [`Error::InterregnumError`] with a specific message.
+    ///
+    /// Interregnum errors are a specific type of error arising from differences
+    /// between types in different versions of Nexus. For example, a param
+    /// struct enum and db struct enum may have the same variants for one
+    /// version of Nexus, but different variants when one of those Nexus servers
+    /// is upgraded. Deserializing from the database in the new Nexus would be
+    /// ok, but in the old Nexus would otherwise be a 500.
+    pub fn interregnum_error(message: &str) -> Error {
+        Error::InterregnumError { internal_message: message.to_owned() }
+    }
```

Comment:
Perhaps it could be something like `type_version_mismatch` or something more explicit?

Reply:
I like `type_version_mismatch`, changed in f3c5ede0